api/src/main/java/org/apache/iceberg/util/CharSequenceWrapper.java

```diff
@@ -67,7 +78,13 @@ public boolean equals(Object other) {
 
   @Override
   public int hashCode() {
-    return JavaHashes.hashCode(wrapped);
+    int hash = hashCode;
```

Comment:
this uses a local `hash` var on purpose due to https://jeremymanson.blogspot.com/2008/12/benign-data-races-in-java.html. Also the `String` class itself uses the same approach when lazily calculating and caching a `hashCode`